File: dongho-kim/5주차/Longest Consecutive Sequence/solution2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        answer = 0

        num_set = set(nums)
        for num in num_set:
            # 시작점인 경우
            if (num - 1) not in num_set:
                curr = num
                length = 1

                while (curr + 1) in num_set:
                    curr += 1
                    length += 1

                answer = max(answer, length)

        return answer

File: pmh/study1/week5/LongestConsecutiveSequence/Solution2.java
package study1.week5.LongestConsecutiveSequence;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;

class Solution2 {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();

        for(int num : nums){
            set.add(num);
        }
        int max =0;
        for (int num : set) {
            if(!set.contains(num-1)){
                int cur = 1;
                while(set.contains(num+1)){
                    cur++;
                    num++;
                    max = Math.max(cur, max);
                }
            }
        }
        return max;
    }
}